share/cmake/modules/FindImath.cmake

```diff
@@ -0,0 +1,222 @@
+            file(STRINGS "${_Imath_CONFIG}" _Imath_VER_SEARCH 
+                REGEX "^[ \t]*#define[ \t]+IMATH_VERSION_STRING[ \t]+\"[.0-9]+\".*$")
+            if(_Imath_VER_SEARCH)
+                string(REGEX REPLACE ".*#define[ \t]+IMATH_VERSION_STRING[ \t]+\"([.0-9]+)\".*" 
+                    "\\1" Imath_VERSION "${_Imath_VER_SEARCH}")
+            endif()
+        elseif(PC_Imath_FOUND)
+            set(Imath_VERSION "${PC_Imath_VERSION}")
+        endif()
+    endif()
+
+    # Override REQUIRED if package can be installed
+    if(OCIO_INSTALL_EXT_PACKAGES STREQUAL MISSING)
+        set(Imath_FIND_REQUIRED FALSE)
+    endif()
+
+    include(FindPackageHandleStandardArgs)
+    find_package_handle_standard_args(Imath
+        REQUIRED_VARS 
+            ${_Imath_REQUIRED_VARS}
+        VERSION_VAR
+            Imath_VERSION
+    )
+endif()
+
+###############################################################################
+### Create target
+
+if (NOT TARGET Imath::Imath)
+    add_library(Imath::Imath UNKNOWN IMPORTED GLOBAL)
+    set(_Imath_TARGET_CREATE TRUE)
+endif()
+
+###############################################################################
+### Install package from source ###
+
+if(NOT Imath_FOUND)
+    include(ExternalProject)
+    include(GNUInstallDirs)
+
+    set(_EXT_DIST_ROOT "${CMAKE_BINARY_DIR}/ext/dist")
+    set(_EXT_BUILD_ROOT "${CMAKE_BINARY_DIR}/ext/build")
+
+    # Set find_package standard args
+    set(Imath_FOUND TRUE)
+    set(Imath_VERSION ${Imath_FIND_VERSION})
+    set(Imath_INCLUDE_DIR "${_EXT_DIST_ROOT}/${CMAKE_INSTALL_INCLUDEDIR}/Imath")
+
+    # Set the expected library name. "_d" is appended to Debug Windows builds 
+    # <= OpenEXR 2.3.0. In newer versions, it is appended to Debug libs on
+    # all platforms.
+    if(BUILD_TYPE_DEBUG)
+        set(_Imath_LIB_SUFFIX "_d")
+    endif()
+
+    set(Imath_LIBRARY
+        "${_EXT_DIST_ROOT}/${CMAKE_INSTALL_LIBDIR}/${CMAKE_STATIC_LIBRARY_PREFIX}Imath-${_Imath_LIB_VER}${_Imath_LIB_SUFFIX}${CMAKE_STATIC_LIBRARY_SUFFIX}")
+
+    if(_Imath_TARGET_CREATE)
+        if(MSVC)
+            set(Imath_CXX_FLAGS "${Imath_CXX_FLAGS} /EHsc")
+        endif()
+
+        string(STRIP "${Imath_CXX_FLAGS}" Imath_CXX_FLAGS)
+ 
+        set(Imath_CMAKE_ARGS
+            ${Imath_CMAKE_ARGS}
+            -DCMAKE_CXX_VISIBILITY_PRESET=${CMAKE_CXX_VISIBILITY_PRESET}
+            -DCMAKE_VISIBILITY_INLINES_HIDDEN=${CMAKE_VISIBILITY_INLINES_HIDDEN}
+            -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+            -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+            -DCMAKE_CXX_FLAGS=${Imath_CXX_FLAGS}
+            -DCMAKE_CXX_STANDARD=${CMAKE_CXX_STANDARD}
+            -DCMAKE_INSTALL_MESSAGE=${CMAKE_INSTALL_MESSAGE}
+            -DCMAKE_INSTALL_PREFIX=${_EXT_DIST_ROOT}
+            -DCMAKE_OBJECT_PATH_MAX=${CMAKE_OBJECT_PATH_MAX}
+            -DBUILD_SHARED_LIBS=OFF
+            -DBUILD_TESTING=OFF
+            -DPYTHON=OFF
+            -DIMATH_HALF_USE_LOOKUP_TABLE=OFF
```

Comment:
`IMATH_HALF_USE_LOOKUP_TABLE` does not exist for Imath 3.0.5!

Reply:
I can move it to a conditional argument for 3.1. That being said, it will result in a CMake warning about an unused argument when present, but should not have any impact on the build otherwise.

Reply:
As we will move to 3.1 at some point (VFX Reference Platform CY2022 is OpenEXR 3.1.x), forget my comment.